pkg/datasources/account_roles.go

```diff
@@ -0,0 +1,94 @@
+package datasources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var accountRolesSchema = map[string]*schema.Schema{
+	"pattern": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Filters the command output by object name.",
+	},
+	"roles": {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "List of all the roles which you can view across your entire account, including the system-defined roles and any custom roles that exist.",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"name": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "Identifier for the role.",
+				},
+				"comment": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The comment on the role",
+				},
+				"owner": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The owner of the role",
+				},
+			},
+		},
+	},
+}
+
+func AccountRoles() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: ReadAccountRoles,
+		Schema:      accountRolesSchema,
+	}
+}
+
+func ReadAccountRoles(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+
+	req := sdk.NewShowRoleRequest()
+	if pattern, ok := d.GetOk("pattern"); ok {
+		req.WithLike(sdk.NewLikeRequest(pattern.(string)))
+	}
+
+	roles, err := client.Roles.Show(ctx, req)
```

Comment:
not that it applies here, but a question I would have is what if there was a describe command? Should we have a separate terraform data source for doing describe command? i.e `data.snowflake_account_role` ?  i think we do this with database and databases

Reply:
I would leave it as is, but please add it to the list of topics we would like to discuss during the planning of supporting GA features.